validation/userValidator.ts

```diff
@@ -0,0 +1,20 @@
+import * as validator from "yup";
+
+const validateLogin = {
+    email: validator.string().email().trim().required(),
```

Comment:
Rather than validateLogin, I suggest defining email, password etc. separately. This way they can be used in other validation files too. You can even move them to a validationDefinition file and import them where they are needed. For example, in validationDefinition.ts:
```
export const emailValidator = yup.email('error message').min().max().trim().required()
```
Then in userValidator.ts:
```
import { emailValidator } from "./valitationDefinition"

export const registerSchema = yup.object({
    email: emailValidator,
    ... other validators
})
```

This makes all of the fields reusable.